utils_for_test.go

```diff
@@ -72,6 +75,20 @@ func OutboundConnection(call *OutboundCall) (*Connection, net.Conn) {
 	return conn, conn.conn
 }
 
+// Return true if the connection matches the passed DiffServ Name.
+func IsTosPriority(c net.Conn, tosPriority tos.ToS) (bool, error) {
+	var connTosPriority int
+	var err error
+
+	if c.RemoteAddr().(*net.TCPAddr).IP.To16() != nil && c.RemoteAddr().(*net.TCPAddr).IP.To4() == nil {
```

Comment:
you can use a complex statement here to help out, e.g

```go
func IsTosPriority(c net.Conn, tosPriority tos.ToS) (bool, error) {
  var tosResolver func () (int, error)
  switch ip := c.RemoteAddr().(*net.TCPAddr).IP; ip {
  case ip.To16() != nil && ip.To4() == nil:
    tosResolver = ipv6.NewConn(c).TrafficClass
  case ip.To4() != nil:
    tosResolver = ipv4.NewConn(c).TOS
  default:
    return tosPriority == 0, nil
  }
  connTosPriority, err := tosResolver()
  return (connTosPriority == int(tosPriority)), err
}
```

excuse any syntax errors, i wrote this in the comment editor.